src/BLSPublicKeyCompendium.sol

```diff
@@ -75,6 +84,24 @@ contract BLSPublicKeyCompendium is IBLSPublicKeyCompendium {
                             VIEW FUNCTIONS
     *******************************************************************************/
 
+    /**
+     * @notice Returns the pubkey of an operator, verifying that the pubkey and its hash are valid
+     * @dev Reverts if the operator has not registered a valid pubkey
+     */
+    function getRegisteredPubkey(address operator) public view returns (BN254.G1Point memory) {
+        BN254.G1Point memory pubkey = operatorToPubkey[operator];
+        bytes32 pubkeyHash = operatorToPubkeyHash[operator];
+
+        require(
+            pubkeyHash != bytes32(0) && BN254.hashG1Point(pubkey) == pubkeyHash, 
+            "BLSPublicKeyCompendium.getRegisteredPubkey: operator is not registered"
+        );
+
+        require(pubkeyHash != ZERO_PK_HASH, "BLSPublicKeyCompendium.getRegisteredPubkey: invalid pubkey");
```

Comment:
don't need this

Reply:
Because it's checked implicitly above?

Reply:
You’re already not allowing zero pubkey to register right?  So it would be impossible for this check to ever fail I think?

Reply:
see [bb4011e](https://github.com/Layr-Labs/eigenlayer-middleware/pull/51/commits/bb4011ede8aa3e0ee29133d38487530f12806056)